sechub-cli/src/daimler.com/sechub/util/filepathmatcher.go

```diff
@@ -3,91 +3,38 @@ package util
-	// - then the last element is inspected to be a file wildcard
-	// - file wildcards are resolved /tried by golang filepath.Match method
-	// - if this is okay we do our special path wildcard matching
-
-
-	/* separate patterns containing path wildcards */
-	subPattern := strings.Split(pattern, "**/")
-//	fmt.Printf("\n%s - seperated into %q", pattern, subPattern)
-	length := len(subPattern)
-	potentialFilePattern := subPattern[length-1]
 
-	/* ------------------------ */
-	/* handle file name pattern */
-	/* ------------------------ */
-	substring := path
-	if strings.Index(potentialFilePattern, "*") != -1 {
-		/* found a file pattern in last entry */
-		/* change length - so slice not changed but last entry no longer handled later */
-		length--
-
-		/* check if file patter is applicable */
-		lastPathIndex := strings.LastIndex(path, "/")
-		fileName := path
-		indexForCut := lastPathIndex
-		if lastPathIndex != -1 {
-			fileName = path[lastPathIndex:]
-		} else {
-			indexForCut = 0
-		}
-		/* we just use the golang integrated filename matcher:*/
-		matches, err := filepath.Match(potentialFilePattern, fileName)
-		if err != nil {
-			fmt.Println(err)
-		}
-		if !matches {
-			return false
-		}
+	// Let's turn the ant style pattern into a regexp:
+	doublestarPatterns := strings.Split(pattern, "**/")
+	//fmt.Printf("\nbefore: %q path: %q", pattern, path)
+	for i, subElement := range doublestarPatterns {
+		// esacpe . with backslash, * to .*
+		doublestarPatterns[i] = strings.Replace(strings.Replace(subElement, ".", "\\.", -1), "*", ".*", -1)
+	}
+	regexpPattern := "^" + strings.Join(doublestarPatterns, ".*/") + "$"
 
-		/* remove filename for further path inspection*/
-		substring = path[indexForCut : len(path)-indexForCut]
+	// add a './' in front of the path except it starts with a '/'
+	// so e.g. "**/.git/*" will also match the current working directory and not only subdirectories
+	if !strings.HasPrefix(path, "/") && strings.Contains(pattern, "/") {
+		path = "./" + path
 	}
 
-	/* -------------------- */
-	/* handle path patterns */
-	/* -------------------- */
-	lastOneWasAsterisk := false
-	for i := 0; i < length; i++ {
-		currentSearch := subPattern[i]
-		if currentSearch == "" {
-			/* this is double asterisk at the begining*/
-			/* so we just accept all here*/
-			lastOneWasAsterisk=true
-			continue
-		}
-		/* search for next occurrence */
-		index := strings.Index(substring, currentSearch)
-		if index == -1 {
-			/* search not found*/
-			return false
-		}
-		if !lastOneWasAsterisk && index != 0 {
-			/* found but not starting with it*/
-			/* when last pattern was not an asterisk ...*/
-			return false
-		}
-		substring = substring[index+len(currentSearch):]
-		/* after first entry we have always path asterisks between - even
-		 * when we got no longer "" between (happens only on first entry)
-		 */
-		lastOneWasAsterisk=true
+	//fmt.Printf("\n after: %q path: %q\n", regexpPattern, path)
```

Comment:
I think the `fmt.Printf` is obsolete.

Reply:
same as for above...

Reply:
Sames as above from my side. Please remove...